core/src/banking_stage/qos_service.rs

```diff
@@ -15,70 +14,25 @@ use {
         saturating_add_assign,
         transaction::{self, SanitizedTransaction, TransactionError},
     },
-    std::{
-        sync::{
-            atomic::{AtomicBool, AtomicU64, Ordering},
-            Arc,
-        },
-        thread::{self, Builder, JoinHandle},
-        time::Duration,
+    std::sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
     },
 };
 
-pub enum QosMetrics {
-    BlockBatchUpdate { slot: Slot },
-}
-
 // QosService is local to each banking thread, each instance of QosService provides services to
 // one banking thread.
-// It hosts a private thread for async metrics reporting, tagged with banking threads ID. Banking
-// thread calls `report_metrics(slot)` at end of `process_and_record_transaction()`, or any time
-// it wants, QosService sends `slot` to reporting thread via channel, signalling stats to be
-// reported if new bank slot has changed.
+// Banking thread calls `report_metrics(slot)` at end of `process_and_record_transaction()`, or any time
+// it wants.
 //
 pub struct QosService {
-    // QosService hosts metrics object and a private reporting thread, as well as sender to
-    // communicate with thread.
-    report_sender: Sender<QosMetrics>,
-    metrics: Arc<QosServiceMetrics>,
-    // metrics reporting runs on a private thread
-    reporting_thread: Option<JoinHandle<()>>,
-    running_flag: Arc<AtomicBool>,
-}
-
-impl Drop for QosService {
-    fn drop(&mut self) {
-        self.running_flag.store(false, Ordering::Relaxed);
-        self.reporting_thread
-            .take()
-            .unwrap()
-            .join()
-            .expect("qos service metrics reporting thread failed to join");
-    }
+    metrics: QosServiceMetrics,
```

Comment:
🤔 fields on these metrics no longer need to be atomic since they aren't shared between threads.

Reply:
@brooksprumo fyi, I took a stab at doing this but it has an unfortunate consequence compared to the atomics. Atomics don't require mutability to be modified, whereas the u64s do.

If we made the change I think it'd force us to make the QosService mutable, and in turn make the Consumer itself mutable.

I plan to leave this as is w/ the atomics - it also leaves us free to share QoS service between non-vote threads in the future (we mostly care about separating non-votes from votes rather than per-thread metrics).